File: eCinema/eCinema.Core/Entities/Reservation.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace eCinema.Core.Entities
{
    public class Reservation : BaseEntity
    {
        public int ID { get; set; }
        public bool isActive { get; set; }

        public string TransactionNumber { get; set; } = null!;

        [ForeignKey(nameof(Show))]
        public int ShowId { get; set; }
        public Show Show { get; set; } = null!;

        [ForeignKey(nameof(User))]
        public int UserId { get; set; }
        public User User { get; set; } = null!;
        public ICollection<ReservationSeat> Seats { get; set; } = new List<ReservationSeat>();
    }
}

File: eCinema/UI/ecinema_mobile/lib/models/reservation.dart
import 'package:ecinema_mobile/models/reservation_seat.dart';
import 'package:ecinema_mobile/models/show.dart';
import 'package:ecinema_mobile/models/user.dart';
import 'package:json_annotation/json_annotation.dart';

part 'reservation.g.dart';

@JsonSerializable()
class Reservation {
  int? id;
  bool? isActive;
  String? transactionNumber;
  int? showId;
  Show? show;
  int? userId;
  User? user;
  List<ReservationSeat>? seats;
  double? totalPrice;

  Reservation(this.id, this.isActive, this.transactionNumber, this.showId, this.show, this.seats, this.userId,
      this.user, this.totalPrice);

  factory Reservation.fromJson(Map<String, dynamic> json) => _$ReservationFromJson(json);

  Map<String, dynamic> toJson() => _$ReservationToJson(this);
}
